File: Common/InMotionGIT.Common.Domain/DataType/Argument.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.Serialization;
using System.Xml.Serialization;

namespace InMotionGIT.Common.Domain.DataType;


[DataContract(Namespace = "urn:InMotionGIT.Common.DataType")]
[Serializable()]
[XmlType(Namespace = "urn:InMotionGIT.Common.DataType")]
[XmlRoot(Namespace = "urn:InMotionGIT.Common.DataType")]
public class Argument
{

    [DataMember()]
    [XmlAttribute()]
    public string Alias { get; set; }

    [DataMember()]
    [XmlAttribute()]
    public string Type { get; set; }

    [DataMember()]
    [XmlAttribute()]
    public string Assemblies { get; set; }

    [DataMember()]
    [XmlAttribute()]
    public string Name { get; set; }

    [DataMember()]
    [XmlAttribute()]
    public bool IsCollection { get; set; }

    [DataMember()]
    [XmlAttribute()]
    public bool FileContent { get; set; }

    public List<Argument> Arguments { get; set; }

    [XmlIgnore()]
    public string FullName
    {
        get
        {
            return string.Format("{0}.{1}", Type, Name);
        }
    }

    [XmlIgnore()]
    public Type RealType
    {
        get
        {
            Type result = null;

            string currentFullName = FullName;

            if (Type.IndexOf("/") > -1)
            {
                currentFullName = Type.Split('/')[Type.Split('/').Length - 1];
            }

            foreach (System.Reflection.Assembly AssemblyItem in AppDomain.CurrentDomain.GetAssemblies())
            {
                if (!AssemblyItem.FullName.StartsWith("inrule.", StringComparison.CurrentCultureIgnoreCase))
                {
                    foreach (Type typeItem in AssemblyItem.GetTypes())
                    {
                        if ((currentFullName ?? "") == (typeItem.FullName ?? ""))
                        {
                            result = typeItem;
                            break;
                        }
                    }

                    if (!(result == null))
                    {
                        break;
                    }
                }
            }

            return result;
        }
    }

    [XmlIgnore()]
    [Browsable(false)]
    public string FormatDBType
    {
        get
        {
            string kind = FullName;

            if (kind.Contains("System."))
            {
                kind = kind.Split('.')[1];
            }

            switch (kind.ToUpper() ?? "")
            {
                case "STRING":
                case "BOOLEAN":
                    {
                        return "VARCHAR(1)";
                    }

                case "INT16":
                case "INT32":
                case "DECIMAL":
                case "INT64":
                    {
                        return "NUMBER";
                    }

                case "DATETIME":
                    {
                        return "DATE";
                    }

                default:
                    {
                        return "VARCHAR(1)";
                    }
            }
        }
    }

}

File: Common/InMotionGIT.Common - Copy/DataType/Argument.vb
Imports System.ComponentModel
Imports System.Runtime.Serialization
Imports System.Xml.Serialization

Namespace DataType

    <DataContract(Namespace:="urn:InMotionGIT.Common.DataType")>
    <Serializable()>
    <XmlType(Namespace:="urn:InMotionGIT.Common.DataType")>
    <XmlRoot(Namespace:="urn:InMotionGIT.Common.DataType")>
    Public Class Argument

        <DataMember()>
        <XmlAttribute()>
        Public Property [Alias] As String

        <DataMember()>
        <XmlAttribute()>
        Public Property Type As String

        <DataMember()>
        <XmlAttribute()>
        Public Property Assemblies As String

        <DataMember()>
        <XmlAttribute()>
        Public Property Name As String

        <DataMember()>
        <XmlAttribute()>
        Public Property IsCollection As Boolean

        <DataMember()>
        <XmlAttribute()>
        Public Property FileContent As Boolean

        Public Property Arguments() As List(Of Argument)

        <XmlIgnore()>
        Public ReadOnly Property FullName As String
            Get
                Return String.Format("{0}.{1}", Type, Name)
            End Get
        End Property

        <XmlIgnore()>
        Public ReadOnly Property RealType() As Type
            Get
                Dim result As Type = Nothing

                Dim currentFullName As String = FullName

                If Type.IndexOf("/") > -1 Then
                    currentFullName = Type.Split("/")(Type.Split("/").Length - 1)
                End If

                For Each AssemblyItem As Reflection.Assembly In AppDomain.CurrentDomain.GetAssemblies
                    If Not AssemblyItem.FullName.StartsWith("inrule.", StringComparison.CurrentCultureIgnoreCase) Then
                        For Each typeItem As Type In AssemblyItem.GetTypes
                            If currentFullName = typeItem.FullName Then
                                result = typeItem
                                Exit For
                            End If
                        Next

                        If Not IsNothing(result) Then
                            Exit For
                        End If
                    End If
                Next

                Return result
            End Get
        End Property

        <XmlIgnore()>
        <BrowsableAttribute(False)>
        Public ReadOnly Property FormatDBType() As String
            Get
                Dim kind As String = FullName

                If kind.Contains("System.") Then
                    kind = kind.Split(".")(1)
                End If

                Select Case kind.ToUpper
                    Case "STRING", "BOOLEAN"
                        Return "VARCHAR(1)"

                    Case "INT16", "INT32", "DECIMAL", "INT64"
                        Return "NUMBER"

                    Case "DATETIME"
                        Return "DATE"

                    Case Else
                        Return "VARCHAR(1)"
                End Select
            End Get
        End Property

    End Class

End Namespace